File: 2016/03/two.go
package main

import (
    "fmt"
    "os"
	"strconv"
    "strings"
)

func main() {
    data, err := os.ReadFile("input.txt")
    if err != nil { panic(err) }

	possible := 0

    input := string(data)
    lines := strings.Split(strings.Trim(input, "\n"), "\r\n")
	lines2 := make([][] string, len(lines))
	for i, line := range lines {
		lines2[i] = strings.FieldsFunc(line, func(c rune) bool { return c == ' ' })
	}

	for i := 0; i < len(lines2); i += 3 {
		for x := 0; x < 3; x++ {
			a, err := strconv.Atoi(lines2[i][x])
			if err != nil { panic(err) }
			b, err := strconv.Atoi(lines2[i+1][x])
			if err != nil { panic(err) }
			c, err := strconv.Atoi(lines2[i+2][x])
			if err != nil { panic(err) }

			if a + b > c && a + c > b && b + c > a { possible++ }
		}
	}
    
    fmt.Println(possible)
}

File: 2016/09/two.py
data = open("input.txt").read().strip()
def dlength(data):
    c = 0
    while data:
        if data[0] == "(":
            s = data[1:].split(")")
            a, b = [int (x) for x in s[0].split("x")]
            c += dlength(")".join(s[1:])[:a]) * b
            data = ")".join(s[1:])[a:]
        else:
            c += 1
            data = data[1:]
    return c
print(dlength(data))
